packages/core/database/src/entity-manager/relations-orderer.ts

```diff
@@ -51,8 +52,16 @@ const sortConnectArray = (connectArr: Link[], initialArr: Link[] = [], strictSor
         needsSorting = true;
       }
 
-      // If the relation is already in the array, throw an error
-      if (mapper[relation.id]) {
+      // Check if the relation already exists in the mapper
+      const existingRelation = mapper[relation.id];
+
+      // Determine if the existing relation has no component or the same component
+      const hasNoComponent = existingRelation && !('__component' in existingRelation);
+      const hasSameComponent =
+        existingRelation && existingRelation.__component === relation.__component;
```

Comment:
could you elaborate this note a bit more? it might be confusing if the dev reading this doesnt know about the __component attribute